Terminal.Gui/Views/TextView.cs

```diff
@@ -6050,13 +6063,15 @@ private void ProcessPaste ()
         Paste ();
     }
 
-    private bool ProcessReturn ()
+    private bool? ProcessReturn ()
     {
         ResetColumnTrack ();
 
         if (!AllowsReturn || _isReadOnly)
         {
-            return false;
+            // By Default pressing ENTER should be ignored (OnAccept will return false or null). Only cancel if the
+            // event was fired and set Cancel = true.
+            return OnAccept () == false;
```

Comment:
I have serious doubts here. If the text is read-only and `OnAccept()` returns `true`, is the text allowed to be changed? In my opinion it should not be allowed, even in an inherited class.

Reply:
Fixed.